oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,92 @@ impl super::Node for WasmNode {
-        entrypoint: "test_function".to_string(),
-    };
-    let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
-    let (_, reader_handle) =
-        runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
-
-    // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
-    // Minimal module:
-    // \0asm - magic
-    // 0x01 - version (in little-endian)
-    let module =
-        wasmi::Module::from_buffer(vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]).unwrap();
-
-    let mut node = WasmNode::new(
-        "test",
-        runtime_ref,
-        TEST_NODE_ID,
-        Arc::new(module),
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use std::env;
+    use wat::parse_file;
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
+        // Minimal module:
+        // \0asm - magic
+        // 0x01 - version (in little-endian)
+        let module = wasmi::Module::from_buffer(buffer).unwrap();
+
+        Box::new(WasmNode::new(
+            "test",
+            runtime_ref,
+            TEST_NODE_ID,
+            Arc::new(module),
+            entrypoint.to_string(),
+            reader_handle,
+        ))
+    }
+
+    fn from_testfile(file_name: &str) -> Vec<u8> {
+        let mut path = env::current_dir().unwrap();
+        path.pop();
+        path.pop();
+        path.push("testdata");
+        path.push(file_name);
+        parse_file(path).unwrap()
+    }
+
+    #[test]
+    fn wasm_with_invalid_args_fails() {
+        let buffer = vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00];
+
+        // Wrong magic: The minimal module does not have the necessary entry point, so it should
+        // fail immediately.
+        let mut node = setup_node(buffer, "test_function");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_module_without_content_fails() {
+        let binary = from_testfile("empty.wat");
+        let mut node = setup_node(binary, "oak_main");
+        assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    }
+
+    #[test]
+    fn wasm_starting_minimal_module_succeeds() {
+        let binary = from_testfile("minimal.wat");
```

Comment:
I think these tests would read better if the wat files were inlined here as strings, since they are pretty minimal, anyways we can do that in the future if we want. Or perhaps for now you could come up with an additional test case in which the module is specified as a string, in addition to the current ones that we can keep as loading the wasm from a file. WDYT?

Reply:
Are you thinking of creating the wasmi module from something like the following?
```
// copied from https://crates.io/crates/wat

let wat = r#"
    (module
        (func $foo)

        (func (export "bar")
            call $foo
        )
    )
"#;
```

Reply:
Exactly.